src/module/actor/base.js

```diff
@@ -938,86 +938,58 @@ class PTUActor extends Actor {
         const struggles = includeStruggles ? (() => {
             const types = Object.keys(CONFIG.PTU.data.typeEffectiveness)
 
+            // Get the data common between all Struggles out of the way first
             const strugglePlusRollOption = this.rollOptions.struggle ? Object.keys(this.rollOptions.struggle).find(o => o.startsWith("skill:")) : false;
-
-            const struggles = types.reduce((arr, type) => {
-                if (this.rollOptions.struggle?.[`${type.toLocaleLowerCase(game.i18n.lang)}`]) {
-                    const rule = this.rules.find(r => !r.ignored && r.key == "RollOption" && r.domain == "struggle" && r.option == type.toLocaleLowerCase(game.i18n.lang));
-                    const strugglePlus = (() => {
-                        if (strugglePlusRollOption) return this.system.skills?.[strugglePlusRollOption.replace("skill:", "")]?.value?.total > 4;
-                        return this.system.skills?.combat?.value?.total > 4;
-                    })();
-                    const moveData = {
+            const isStrugglePlus = (() => {
+                if (strugglePlusRollOption) return this.system.skills?.[strugglePlusRollOption.replace("skill:", "")]?.value?.total > 4;
+                return this.system.skills?.combat?.value?.total > 4;
+            })();
+            const constructStruggleItem = (type, category, range, ptuFlags = {}, isRangedStruggle = false) => {
+                ptuFlags = ptuFlags ? ptuFlags : {}
+                return new Item.implementation({
                         name: `Struggle (${type})`,
                         type: "move",
                         img: CONFIG.PTU.data.typeEffectiveness[type].images.icon,
                         system: {
-                            ac: strugglePlus ? 3 : 4,
-                            damageBase: strugglePlus ? 5 : 4,
+                            ac: isStrugglePlus ? 3 : 4,
+                            damageBase: isStrugglePlus ? 5 : 4,
                             stab: false,
-                            type: type,
                             frequency: "At-Will",
-                            isStruggle: true
+                            isStruggle: true,
+                            isRangedStruggle: isRangedStruggle,
+                            category: category,
+                            range: range,
+                            type: type
                         },
                         flags: {
-                            ptu: {}
+                            ptu: ptuFlags
                         }
+                    },
+                    {
+                        parent: this,
+                        temporary: true
+                    })
+            }
+            const struggles = types.reduce((arr, type) => {
+                if (this.rollOptions.struggle?.[`${type.toLocaleLowerCase(game.i18n.lang)}`]) {
```

Comment:
You're toLowerCaseing a lot in this for loop so here it'd be worth to store a lower case copy.